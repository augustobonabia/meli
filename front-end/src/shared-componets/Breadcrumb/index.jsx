import React from 'react';
import PropTypes, { string } from 'prop-types';
import './index.scss';

function Breadcrumb(props) {
  const { categories } = props;

  const renderSeparator = (categoryIndex) => {
    if (categoryIndex) {
      return <span className="separator">{'>'}</span>;
    }

    return null;
  };

  const renderCategoryName = (categoryName, categoryIndex) => {
    let className = 'category-name';

    if (categoryIndex + 1 === categories.length) {
      className += ' last-in-path';
    }

    return <span className={className}>{categoryName}</span>;
  };

  return (
    <nav className="breadcrumb">
      {
        categories.map((category, categoryIndex) => (
          // eslint-disable-next-line react/no-array-index-key
          <span key={categoryIndex}>
            {renderSeparator(categoryIndex)}
            {renderCategoryName(category, categoryIndex)}
          </span>
        ))
      }
    </nav>
  );
}

Breadcrumb.propTypes = {
  categories: PropTypes.arrayOf(string).isRequired,
};

export default Breadcrumb;
