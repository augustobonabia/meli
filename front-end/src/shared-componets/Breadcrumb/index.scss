@import '../../core/variables.scss';

nav.no-breadcrumb {
  height: $small-space;
}

nav.breadcrumb {
  color: $secondary-text-color;
  font-size: $breadcumb-font-size;
  margin: $small-space 0;

  span.separator {
    padding: 0 $tiny-space;
  }

  span.category-name.last-in-path {
    font-weight: bold;
  }
}