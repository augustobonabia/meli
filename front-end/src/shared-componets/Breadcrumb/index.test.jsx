import React from 'react';
import Breadcrumb from './index';
import renderer from 'react-test-renderer';

it('El breadcrumb se renderiza correctamente', () => {
  const categories = [
    'Animales y Mascotas',
    'Perros',
    'Perros de Raza',
  ];

  const tree = renderer
    .create(<Breadcrumb categories={categories} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
