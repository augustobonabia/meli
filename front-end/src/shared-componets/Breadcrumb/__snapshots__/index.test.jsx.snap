// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`El breadcrumb se renderiza correctamente 1`] = `
<nav
  className="breadcrumb"
>
  <span>
    <span
      className="category-name"
    >
      Animales y Mascotas
    </span>
  </span>
  <span>
    <span
      className="separator"
    >
      &gt;
    </span>
    <span
      className="category-name"
    >
      Perros
    </span>
  </span>
  <span>
    <span
      className="separator"
    >
      &gt;
    </span>
    <span
      className="category-name last-in-path"
    >
      Perros de Raza
    </span>
  </span>
</nav>
`;
